SOC-3105 Post a welcome message from Fandom staff after migrating forums to discussions

jsutterfield opened the pull request:
Ticket: https://wikia-inc.atlassian.net/browse/SOC-3105

This PR updates the script used to migrate empty forums to discussion communities. Now, following the migration, we create a post from a staff member in the language of the community welcoming them to discussions.

Commit:
SOC-3105 add localization files

Grunny committed:
Fix typo in file name

Commit:
Merge branch 'dev' into SOC-3105

Commit:
SOC-3105 post welcome message from staff after discussion migration

Commit:
SOC-3105 post welcome message from staff after discussion migration

Commit:
Merge branch 'SOC-3105' of github.com:Wikia/app into SOC-3105

Commit:
SOC-3105 formatting changes

Commit:
SOC3105 use trimmed language for staff map and lang ket for message

Commit:
SOC-3105 add constant, rearrange imports

WikiaTech commented on extensions/wikia/Discussions/Discussions.setup.php:
```diff
@@ -16,15 +16,18 @@
+$wgExtensionMessagesFiles['StaffWelcomePost'] = $dir . 'i18n/StaffWelcomePost.i18n.php';
 
 // permissions
 $wgAvailableRights[] = 'specialdiscussions';
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "specialdiscussions" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/Discussions/maintenance/StaffWelcomePoster.class.php:
```diff
@@ -0,0 +1,59 @@
+		$staffId = $this->getStaffFromLang( $trimmedLang );
+		$message = $this->getMessage( $trimmedLang );
+
+		$success = $this->threadCreator->create( $staffId, $siteId,  $message );
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$success". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)

WikiaTech commented on extensions/wikia/Discussions/maintenance/StaffWelcomePoster.class.php:
```diff
@@ -0,0 +1,59 @@
+
+	const LANG_TO_STAFF_MAP = [
+		'de' => 24961680, // ForestFairy
+		'en' => 26339491, // Mira Laime
```
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)

WikiaTech commented:
SonarQube analysis reported 4 issues
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 2 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 2 info

Watch the comments in this conversation to review them.

WikiaTech commented on extensions/wikia/Discussions/maintenance/StaffWelcomePoster.class.php:
```diff
@@ -0,0 +1,59 @@
+		'ko' => 24883131, // Miri-Nae
+		'nl' => 4142476,  // Yatalu
+		'pl' => 1117661,  // Nanaki
+		'pt' => 24005296, // Macherie ana
```
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)

Commit:
SOC-3105 remove extra newline

garthwebb commented on extensions/wikia/Discussions/api/ThreadCreator.class.php:
```diff
@@ -0,0 +1,58 @@
+			// Currently only creates thread in "General" category of the Discussion.
+			$api->createThread( $siteId, $siteId,  $postInput );
+			$success = true;
+		} catch ( ApiException  $e ) {
```
I've found that stringifying the exception `$e` in this way doesn't always yield the best results, giving a generic message rather than the one sent back by the client.  I'd recommend something like:

``` php
$response = $e->getResponseObject();
/** @var \Swagger\Client\Discussion\Models\HalProblem $response */
if ( $response instanceof \Swagger\Client\Discussion\Models\HalProblem ) {
    $detail = $response->getDetail();
} else {
    $detail = $e->getMessage();
}
$this->logError( $siteId, $userId, $body, $detail );
```

jsutterfield replied:
👍  https://github.com/Wikia/app/pull/11632/commits/a8339f9be9ea05ca970481dc917d85fec3201106

garthwebb commented on extensions/wikia/Discussions/maintenance/StaffWelcomePoster.class.php:
```diff
@@ -0,0 +1,59 @@
+	}
+
+	private function getStaffFromLang( string $language ): int {
+		$staffId = self::LANG_TO_STAFF_MAP[$language];
```
This will throw an error for any language that doesn't exist in our core language list.  It looks like the `??` operator protects from that, so this would work:

``` php
return self::LANG_TO_STAFF_MAP[$language] ?? self::LANG_TO_STAFF_MAP[self::DEFAULT_LANG];
```

jsutterfield replied:
Ah! Good catch, I'll update.

Commit:
SOC-3105 avoid NPE

Commit:
SOC-3105 provide better context when logging

Commit:
SOC-3105 download translations

Commit:
SOC-3105 only use simplified chinese

Commit:
SOC-3105 Change that will be really useful when we run this.

Looks like the getDB call can fail and halt the whole program if the dbname doesn't exist (happens in dev, could happen in prod)

Commit:
SOC-3105 fix pl i18n string, allow for zh-hans and zh-hant

WikiaTech commented on extensions/wikia/Discussions/maintenance/StaffWelcomePoster.class.php:
```diff
@@ -0,0 +1,61 @@
+
+	const LANG_TO_STAFF_MAP = [
+		'de' => 24961680,      // ForestFairy
+		'en' => 26339491,      // Mira Laime
```
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)

WikiaTech commented on extensions/wikia/Discussions/maintenance/StaffWelcomePoster.class.php:
```diff
@@ -0,0 +1,61 @@
+		'ko' => 24883131,      // Miri-Nae
+		'nl' => 4142476,       // Yatalu
+		'pl' => 1117661,       // Nanaki
+		'pt' => 24005296,      // Macherie ana
```
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)

Commit:
SOC-3105 map dialects of chinese to zh-hans or zh-hant

WikiaTech commented on extensions/wikia/Discussions/maintenance/StaffWelcomePoster.class.php:
```diff
@@ -0,0 +1,83 @@
+		'pt' => 24005296,      // Macherie ana
+		'ru' => 26457441,      // Vlazovskiy
+		'vi' => 26041741,      // KhangND
+		'zh-hans' => 11909873, // Cal-Boy
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "zh-hans" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/Discussions/maintenance/StaffWelcomePoster.class.php:
```diff
@@ -0,0 +1,83 @@
+		'ru' => 26457441,      // Vlazovskiy
+		'vi' => 26041741,      // KhangND
+		'zh-hans' => 11909873, // Cal-Boy
+		'zh-hant' => 56584     // Ffaarr
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "zh-hant" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)

WikiaTech commented on extensions/wikia/Discussions/maintenance/StaffWelcomePoster.class.php:
```diff
@@ -0,0 +1,83 @@
+		$staffId = $this->getStaffFromLang( $transformedLang );
+		$message = $this->getMessage( $transformedLang );
+
+		$success = $this->threadCreator->create( $staffId, $siteId,  $message );
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$success". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)

WikiaTech commented:
SonarQube analysis reported 6 issues
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 4 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 2 info

Watch the comments in this conversation to review them.

Commit:
SOC-3105 update mTouched after deleting a user's avatar

WikiaTech commented:
SonarQube analysis reported 182 issues
* ![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) 4 critical
* ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) 66 major
* ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 74 minor
* ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 38 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) [User.php#L961](https://github.com/Wikia/app/blob/29b53b443f753797352a4496be5e47c512ef6f62/includes/User.php#L961): Remove this hard-coded password. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2068)
1. ![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) [User.php#L2195](https://github.com/Wikia/app/blob/29b53b443f753797352a4496be5e47c512ef6f62/includes/User.php#L2195): Remove this hard-coded password. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2068)
1. ![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) [User.php#L2199](https://github.com/Wikia/app/blob/29b53b443f753797352a4496be5e47c512ef6f62/includes/User.php#L2199): Remove this hard-coded password. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2068)
1. ![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) [User.php#L3231](https://github.com/Wikia/app/blob/29b53b443f753797352a4496be5e47c512ef6f62/includes/User.php#L3231): Remove this hard-coded password. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2068)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [UserProfilePageController.class.php](https://github.com/Wikia/app/blob/29b53b443f753797352a4496be5e47c512ef6f62/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php): File "UserProfilePageController.class.php" has 1019 lines, which is greater than 1000 authorized. Split it into smaller files. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS104)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [UserProfilePageController.class.php#L6](https://github.com/Wikia/app/blob/29b53b443f753797352a4496be5e47c512ef6f62/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L6): Class "UserProfilePageController" has 26 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [UserProfilePageController.class.php#L61](https://github.com/Wikia/app/blob/29b53b443f753797352a4496be5e47c512ef6f62/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L61): Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [UserProfilePageController.class.php#L160](https://github.com/Wikia/app/blob/29b53b443f753797352a4496be5e47c512ef6f62/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L160): The Cyclomatic Complexity of this function "renderActionButton" is 29 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [UserProfilePageController.class.php#L413](https://github.com/Wikia/app/blob/29b53b443f753797352a4496be5e47c512ef6f62/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L413): Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. ![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) [UserProfilePageController.class.php#L688](https://github.com/Wikia/app/blob/29b53b443f753797352a4496be5e47c512ef6f62/extensions/wikia/UserProfilePageV3/UserProfilePageController.class.php#L688): Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)

Commit:
Revert "SOC-3105 update mTouched after deleting a user's avatar"

This reverts commit 29b53b443f753797352a4496be5e47c512ef6f62.

WikiaTech commented:
SonarQube analysis reported 6 issues
- ![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) 4 minor
- ![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) 2 info

Watch the comments in this conversation to review them.

Commit:
SOC-3105 Fix spacing and make sure args are passed in the right order

WikiaTech commented on extensions/wikia/Discussions/maintenance/StaffWelcomePoster.class.php:
```diff
@@ -0,0 +1,83 @@
+		$staffId = $this->getStaffFromLang( $transformedLang );
+		$message = $this->getMessage( $transformedLang );
+
+		$success = $this->threadCreator->create( $staffId, $siteId, $message );
```
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "$success". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1488)
